### PR TITLE
Features/#357 make independent kmean from e hv clustering

### PR DESCRIPTION
**Pull request by CarlosEpia**
This PR deals with issues #357 and #333. The previous PR was deleted since this branch makes significant changes. In case of testing the eHV and kmean clustering while using the SH db, take into account that there are about 19 eHV buses.

### CODE REVIEW

File: etrago/cluster/networkclustering.py [hunk 1]
```diff
@@ -178,25 +300,25 @@ def cluster_on_extra_high_voltage(network, busmap, with_time=True):
...
     mask = lines.bus0.isin(buses.index)
     lines = lines.loc[mask, :]
-    
+
     # keep attached links
     links = network.links.copy()
-    dc_links = links[links['carrier'] == 'AC']
-    links = links[links['carrier'] != 'AC']
+    dc_links = links[links["carrier"] == "AC"]
+    links = links[links["carrier"] != "AC"]
```
Comment (ClaraBuettner) [inline etrago/cluster/networkclustering.py:321]: DC-links have carrier =  'DC'

File: etrago/cluster/networkclustering.py [hunk 2]
```diff
@@ -215,74 +337,57 @@ def cluster_on_extra_high_voltage(network, busmap, with_time=True):
...
-                'p_min_pu':np.min,
-                'p_max_pu': np.max,
-                'marginal_cost': np.mean,
-                'terrain_factor': _make_consense_links,
-                'p_nom_opt': np.mean,
-                'country': _make_consense_links}
-    
-    network_c.links = network2.links.groupby(['bus0', 'bus1', 'carrier']).agg(strategies)
+    network2.links.dropna(subset=["bus0", "bus1"], inplace=True)
+    network2.links["topo"] = np.nan
```
Comment (ClaraBuettner) [inline etrago/cluster/networkclustering.py:374]: Some links (e.g. heat pumps) and lines have time series values. Is this considered in the clustering? I couldn't find it in your function.

File: etrago/cluster/networkclustering.py (hunk 1 above)
Comment (CarlosEpia) [reply etrago/cluster/networkclustering.py:327]: Solved in commit 98adb4a5b1c29ffcac1441e9081caec342a31caf.

File: etrago/cluster/networkclustering.py (hunk 2 above)
Comment (CarlosEpia) [reply etrago/cluster/networkclustering.py:377]: This problem will be fixed with issue #366

**Review by ClaraBuettner — APPROVED** [review]

**Pull request merged**